src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/Option2.cs

```diff
@@ -26,6 +26,10 @@ internal interface ISingleValuedOption : IOption2
         /// Note that this property is not (and should not be) used for computing option values or storing options.
         /// </remarks>
         public string? LanguageName { get; }
+
+#if !CODE_STYLE
+        void WriteToGlobalOptionService(IGlobalOptionService globalOptionService, string value);
```

Comment:
If anyone tells me a better idea of how could I write a string value to the IGlobalOptionService with generics I would be happy. Since `void SetGlobalOption<T>(Option2<T> option, T value)`, takes `T`, if not doing so I need to list all different types of options.

Reply:
https://github.com/dotnet/roslyn/blob/475293444be097e8a7cab8658b99d3c706b6e8e6/src/Workspaces/Core/Portable/Options/IGlobalOptionService.cs#L48